gateway/webhook/webhook.go

```diff
@@ -297,7 +310,14 @@ func (bt *batchWebhookTransformerT) batchTransformLoop() {
 				continue
 			}
 
-			payloadArr = append(payloadArr, body)
+			payload, err := sourceTransformAdapter.getTransformerEvent(req.authContext.SourceID, body)
```

Comment:
Source is a huge object. It has source definition, list of destinations attached and each destination has destination config which is also huge. Instead of copying and passing it around for all requests, why not get it only when required?

Reply:
authContext is a pointer, the source object can be passed by reference too. We just resolve things once and we are done, instead of constantly looking for them and exposing new methods/ dealing with locks